pkg/policy/policy_controller.go

```diff
@@ -277,17 +276,18 @@ func (pc *PolicyController) deletePolicy(obj interface{}) {
 
 	logger.Info("policy deleted", "uid", p.UID, "kind", "ClusterPolicy", "name", p.Name)
 
-	// we process policies that are not set of background processing
-	// as we need to clean up GRs when a policy is deleted
-	// skip generate policies with clone
-	rules := autogen.ComputeRules(p)
+	pc.enqueueRCRDeletedPolicy(p.Name)
```

Comment:
I don't get the logic here.
Shouldn't we check the state in UR controller instead ?